File: Medium/1441. Build an Array With Stack Operations/1441. Build an Array With Stack Operations.js
/**
 * @param {number[]} target
 * @param {number} n
 * @return {string[]}
 */
var buildArray = function (target, n) {
    const result = [];
    let current = 1;

    for (const num of target) {
        while (current < num) {
            result.push('Push');
            result.push('Pop');
            current++;
        }

        result.push('Push');
        current++;
    }

    return result;
};

File: Medium/1441. Build an Array With Stack Operations/1441. Build an Array With Stack Operations.rs
impl Solution {
    pub fn build_array(target: Vec<i32>, n: i32) -> Vec<String> {
        let mut result = Vec::new();
        let mut current = 1;

        for &num in &target {
            while current < num {
                result.push("Push".to_string());
                result.push("Pop".to_string());
                current += 1;
            }

            result.push("Push".to_string());
            current += 1;
        }

        result
    }
}
